modules/programs/vscode.nix

```diff
@@ -211,12 +247,14 @@ in {
           else
             builtins.attrNames (builtins.readDir (ext + "/${subDir}")));
       in if cfg.mutableExtensionsDir then
-        mkMerge (concatMap toPaths cfg.extensions)
+        mkMerge (concatMap toPaths cfg.extensions
+        ++ [{ "${extensionPath}/extensions.json".text = extensionJson; }])
```

Comment:
I can confirm that this prevents vscode from installing extensions manually. Maybe this should be an `home.activation` script that writes `extensions.json` if it doesn't exist?

Reply:
Please do!

Reply:
Alternatively, it seems like we can make vscode regenerate the file itself with

```
rm -f ${extensionPath}/{extensions.json,.init-default-profile-extensions}
${vscodeExecutable} --list-extensions > /dev/null
```

This is probably better since it will also remove extensions.

Reply:
This seems to work well, I've pushed it to your branch.